client/post-editor/editor-drawer/term-selector.jsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import without from 'lodash/without';
+import isEqual from 'lodash/isEqual';
+import clone from 'lodash/clone';
+
+/**
+ * Internal dependencies
+ */
+import TermTreeSelector from 'my-sites/term-tree-selector';
+import PostActions from 'lib/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class EditorDrawerTermSelector extends Component {
+	static propTypes = {
+		siteId: React.PropTypes.number,
+		selectedTermIds: React.PropTypes.array,
+		taxonomyName: React.PropTypes.string
+	};
+
+	constructor( props ) {
+		super( props );
+		this.boundOnTermsChange = this.onTermsChange.bind( this );
+
+		// This does not seem like a solid approach here
+		this.state = {
+			selectedIds: props.selectedTermIds
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps.selectedTermIds, this.state.selectedIds ) ) {
+			this.setState( { selectedIds: nextProps.selectedTermIds } );
+		}
+	}
+
+	onTermsChange( selectedTerm ) {
+		const { taxonomyName } = this.props;
+		let termIds = clone( this.state.selectedIds ) || [];
```

Comment:
When will `selectedIds` be falsey to lead to the fallback? Looks from `<EditorDrawerTaxonomies />` that this won't happen. Otherwise, could we specify a default prop? 

In general, you could drop the clone in favor of:

```js
let termIds = [ ...this.state.selectedIds ];
```

But here, even better, you can avoid an unnecessary clone in the case that the term is already selected (`without` doesn't mutate) by using `Array#concat`:

```js
let termIds = this.state.selectedIds;
if ( -1 === termIds.indexOf( selectedTerm.ID ) ) {
	termIds = termIds.concat( selectedTerm.ID );
} else {
	termIds = without( termIds, selectedTerm.ID );
}